lib/file.js

```diff
@@ -235,24 +235,22 @@ class File extends EventEmitter {
         if (chunkSize < maxChunkSize) {
           chunkSize = chunkSize + chunkSizeIncrement
         }
-
-        activeStreams += 1
-        if (activeStreams < maxConnections) {
-          setTimeout(getChunk, 1000)
-        }
       }
 
       if (maxConnections === 1) {
         const r = requestModule(response.g + '/' + apiStart + '-' + (end - 1))
         r.on('error', handleErrors)
+        r.on('response', handleErrors)
```

Comment:
Well, I see that there are some bugs in error handling: this part should be `.on('response', ...)` as `handleErrors` actually handle HTTP errors, not connection errors. In the other hand the function used in `.on('error', ...)` should handle other kind of errors (DNS resolution, TCP level errors, or actual HTTP parse errors / [source](https://nodejs.org/api/http.html)) so it was a mistake of mine using `handleErrors` for that, other mistake was choosing this name for the function.

Reply:
Variable activeStreams has been removed, connection errors have been passed to stream error.